edg_acoustics/acoustics_simulation.py

```diff
@@ -1012,7 +1016,90 @@ def init_Flux(self, Flux: edg_acoustics.Flux):
+
+        
+        Args:
+
+        Returns:
+        """
+        self.time_integrator = time_integrator
+
+
+    def time_integration(self, rec: numpy.ndarray, **kwargs):
+        """
+        perform the time integration
+
+        
+        Args:
+
+        kwargs:
+            n_time_steps (int): number of time steps to compute.
+            total_time (float): total simulation time to compute, determines the number of time steps given the current time step.
+
+        Returns:
+        """
+
+        # Process optional input arguments
+        if ("n_time_steps" in kwargs and "total_time" in kwargs):
+            # Not possible to input both number of steps and total simulation time
+            raise ValueError("Set only n_time_steps or total_time, do not set both...")
+        
+        elif ("n_time_steps" in kwargs):
+            # Directly use the number of timesteps
+            self.Ntimesteps = kwargs['n_time_steps'] 
+
+        elif ("total_time" in kwargs):
+            # Compute the number of time steps from the total simulation time and the time step size of the time integrator
+            total_time = kwargs['total_time']
+            self.Ntimesteps = math.floor(total_time / self.time_integrator.dt)  # apalha: I would make this math.ceil, because you want to simulate a bit more and not less
+
+        else:
+            raise ValueError("You need to set n_time_steps or total_time...")
+        
+
+        
+
+        print(f"Total simulation time is {total_time}")
+        print(f"Total number of simulation steps is {self.Ntimesteps}")
+
+        self.prec = numpy.zeros([rec.shape[1], self.Ntimesteps])
+        self.sampleWeight, self.nodeindex = AcousticsSimulation.sample3D(self.mesh.EToV, self.mesh.vertices, self.xyz, rec, self.Nx)
+
+        # Reset the time step variables
+        self.P = self.P0.copy()
+        self.Vx = self.Vx0.copy()
+        self.Vy = self.Vy0.copy()
+        self.Vz = self.Vz0.copy()
+
+        # Step the solution
+        for StepIndex in range(self.Ntimesteps):
+            print(f"Current/Total step {StepIndex+1}/{self.Ntimesteps}")
+            print(f"Current/Total time {self.time_integrator.dt * StepIndex}/{total_time}")
+            # print(f"outside, self.P ID {id(self.P)}, self.P0 ID {id(self.P0)}")
+            # print(f"outside, self.BC.BCvar ID {id(self.BC.BCvar)}")
+
+            # Reset the initial condition to prepare for the new time step
+            # For the first step this does nothing, but for the next steps it does the work
+            # Resetting before is better than after because in this way the previous step and
+            # current step solutions are available, which can always be useful.
+            self.P0 = self.P.copy()
+            self.Vx0 = self.Vx.copy()
+            self.Vy0 = self.Vy.copy()
+            self.Vz0 = self.Vz.copy()
+
+            self.P, self.Vx, self.Vy, self.Vz = self.time_integrator.step_dt_new(self.P0, self.Vx0, self.Vy0, self.Vz0) 
```

Comment:
I see your point. But, apart from P0 and V0, we also need to include self.BC, since those auxiliary varibles also need to be time integrated.

Reply:
Yes, I did not change that part.

I am a bit uncertain about the BCs. Not with removing them from the integrator, but how they are actually implemented. But we can discuss later.